fail2ban: bump to 1.1.0

HiHat opened the pull request:
## 📦 Package Details

**Maintainer:** 

**Description:**
Bump fail2ban version to 1.1.0

fail2ban changes:
- nftables support (iptables dependency removed)
- python3 support (old package patches removed)
- filter.d/dropbear.conf failregex extended to match different format of "Exit before auth" message

Fixes: [fail2ban: very old version #23015](https://github.com/openwrt/packages/issues/23015)

---

## 🧪 Run Testing Details
* **OpenWrt Version:** OpenWrt 24.10
* **OpenWrt Target/Subtarget:** mediatek/filogic
* **OpenWrt Device:** none
---

## ✅ Formalities

- [x] I have reviewed the [CONTRIBUTING.md](../CONTRIBUTING.md) file for detailed contributing guidelines.

### If your PR contains a patch:
* [x]  It can be applied using `git am`
* [x]  It has been refreshed to avoid offsets, fuzzes, etc., using
  ```shell
  make package/<your-package>/refresh V=s
  ```
* [x]  It is structured in a way that it is potentially upstreamable
  (e.g., subject line, commit description, etc.)
  We must try to upstream patches to reduce maintenance burden.

HiHat commented:
@erdoukki, kindly ask you to avpprove!

wehagy commented:
Please, follow the [CONTRIBUTING.md](https://github.com/openwrt/packages/blob/master/CONTRIBUTING.md) and [Submission Guidelines](https://openwrt.org/submitting-patches#submission_guidelines).

HiHat commented:
@erdoukki, kindly ask you to approve!

wehagy commented:
Plese one more time, follow the [CONTRIBUTING.md](https://github.com/openwrt/packages/blob/master/CONTRIBUTING.md) and [Submission Guidelines](https://openwrt.org/submitting-patches#submission_guidelines). Otherwise the changes are unlikely to be reviewed and merged by the package maintainers.

- The commit subject must be less than 50 characters long.
  - `fail2ban: bump to 1.1.0` is good enough.
- Use the commit description to describe your changes, it must be less than 75 characters per line, clearly explain changes.
  - Why remove these patches?
  - Why add these patches? Where is the patch header?
  - Why were dependencies removed?
  - Add Fixes: https://github.com/openwrt/packages/issues/23015 to the description.
  - Is the `iptables` dependency necessary? Shouldn't `fail2ban` support `nftables`?
- Remove the @erdoukki as maintainer, unresponsive/unreachable.

HiHat commented:
> * The commit subject must be less than 50 characters long.
>   * `fail2ban: bump to 1.1.0` is good enough.
> * Use the commit description to describe your changes, it must be less than 75 characters per line, clearly explain changes.

Fixed.

>   * Why remove these patches?
>   * Why were dependencies removed?
>   * Add Fixes: [fail2ban: very old version #23015](https://github.com/openwrt/packages/issues/23015) to the description.
>   * Is the `iptables` dependency necessary? Shouldn't `fail2ban` support `nftables`?
> * Remove the @erdoukki as maintainer, unresponsive/unreachable.

Fixed.
Please check!

GeorgeSapkin commented on net/fail2ban/patches/030-dropbear-regex-fix.patch:
This patch still has no patch header. Where is this coming from?

HiHat replied:
> This patch still has no patch header. Where is this coming from?

Fixed!
Patch was originated from upstream repo and modified by me.

HiHat commented:
I have no idea how to fix it:
```
Testing package libsqlite3-0 in version 3.50.4 from libs/sqlite3
No pre-test.sh script available
(1/1) Installing libsqlite3-0 (3.50.4-r1)
Bus error (core dumped)
Error: Process completed with exit code 138.
```

GeorgeSapkin commented:
I think the failed test is a CI issue that has nothing to do with the PR.

As for the PR itself, I'm hesitant about merging this, since I don't use fail2ban and a custom patch doesn't sit well with me. Can it be upstreamed? What does ASUSWRT have to do with OpenWrt? As the PR template says:

> We must try to upstream patches to reduce maintenance burden.

wehagy commented:
> I think the failed test is a CI issue that has nothing to do with the PR.

Yep, this error can be ignored.

> As for the PR itself, I'm hesitant about merging this, since I don't use fail2ban 

I'm in the same boat. This update appears to be huge, so I plan to test it myself if no one else tests it before me. But first, I need to understand how the package works.

> custom patch doesn't sit well with me. Can it be upstreamed? What does ASUSWRT have to do with OpenWrt? As the PR template says:

The patch is a backport, so it's fine.

HiHat commented on net/fail2ban/Makefile:
```diff
@@ -6,30 +6,27 @@
-PKG_HASH:=383108e5f8644cefb288537950923b7520f642e7e114efb843f6e7ea9268b1e0
+PKG_HASH:=474fcc25afdaf929c74329d1e4d24420caabeea1ef2e041a267ce19269570bae
 
-PKG_MAINTAINER:=Gerald Kerma <gandalf@gk2.net>
```
Fixed!

wehagy commented on net/fail2ban/patches/010-dropbear-regex-fix.patch:
```diff
@@ -0,0 +1,24 @@
+From a796cc9b91656721fee0d1904911101c678452ad Mon Sep 17 00:00:00 2001
+From: sebres <info@sebres.de>
+Date: Fri, 27 Dec 2024 16:43:33 +0100
+Subject: [PATCH] `filter.d/dropbear.conf`: failregex extended to match
+ different format of "Exit before auth" message; closes gh-3791
+
+Signed-off-by: Andrey Zotikov <andrey.zotikov@gmail.com>
+
+--- a/config/filter.d/dropbear.conf
++++ b/config/filter.d/dropbear.conf
+@@ -25,9 +25,10 @@ _daemon = dropbear
+ 
+ prefregex = ^%(__prefix_line)s<F-CONTENT>(?:[Ll]ogin|[Bb]ad|[Ee]xit).+</F-CONTENT>$
+ 
+-failregex = ^[Ll]ogin attempt for nonexistent user ('.*' )?from <HOST>:\d+$
+-            ^[Bb]ad (PAM )?password attempt for .+ from <HOST>(:\d+)?$
+-            ^[Ee]xit before auth \(user '.+', \d+ fails\): Max auth tries reached - user '.+' from <HOST>:\d+\s*$
++failregex = ^[Ll]ogin attempt for nonexistent user (?:'<F-USER>.*</F-USER>' )?from <HOST>:\d+$
++            ^[Bb]ad (?:PAM )?password attempt for '<F-USER>.+</F-USER>' from <HOST>(?::\d+)?$
++            ^[Ee]xit before auth from \<?<ADDR>:\d+\>?: (?:\([^\)]*\): )?Max auth tries reached - user '<F-USER>.+</F-USER>'\s*$
++            ^[Ee]xit before auth \(user '.+', \d+ fails\): Max auth tries reached - user '<F-USER>.+</F-USER>' from <HOST>:\d+\s*$
+ 
+ ignoreregex = 
+ 
```
> I will try to adapt the configuration for OpenWrt, but using it may lead to flash damage due to sqlite.

Please explain.

HiHat replied:
> Please explain.

According to https://github.com/peci1/fail2ban_openwrt:

_Fail2ban stores a database of already banned IP addresses. It is best if this file survives restarts of the router. By default, it is put in /var/lib/fail2ban/fail2ban.sqlite3. If your /var directory resides in memory only, you might want to change the config in /etc/config/fail2ban to point to a persistent place. But be aware that if your router's system is on an an eMMC flash memory, it is possibly not meant to keep often-changing files and you should rather connect an external drive and set it as the location for fail2ban database, otherwise it can destroy your router._

wehagy replied:
I'm not exactly sure how the patches affect this, but, in any case, this is the default behavior.

wehagy commented:
> Or fail2ban needs to be modified to process logread logs.

This is the way; adding more dependencies can be a problem for some targets.
I thought about adding a check to the init script: if the system is already saving the log to a file, use it; otherwise write the log to `/var/log/system`.

> I forgot to remove it!

To be honest, it is really rare for an external contributor to remove packages…
